docs/CUSTOM_RULES_101.md

```diff
@@ -0,0 +1,529 @@
+    Type returnType = symbol.returnType().type();
+    reportIssue(method.simpleName(), "Never do that!");
+  }
+}
+```
+
+Since the rule should only raise an issue when these two types are the same, we then simply test if the return type is the same as the type of the first parameter using method `is(String fullyQualifiedName)`, provided through the `Type` class, before raising the issue.
+
+```java
+@Override
+public void visitNode(Tree tree) {
+  MethodTree method = (MethodTree) tree;
+  if (method.parameters().size() == 1) {
+    MethodSymbol symbol = method.symbol();
+    Type firstParameterType = symbol.parameterTypes().get(0);
+    Type returnType = symbol.returnType().type();
+    if (returnType.is(firstParameterType.fullyQualifiedName())) {
+      reportIssue(method.simpleName(), "Never do that!");
+    }
+  }
+}
+```
+
+Now, **execute the test** class again.
+
+Test passed? If not, then check if you somehow missed a step.
+
+If it passed...
+
+> ***
+> :tada: **Congratulations!** :confetti_ball:
+>
+> [*You implemented your first custom rule for the SonarQube Java Analyzer!*](resources/success.jpg)
+> ***
+
+## Registering the rule in the custom plugin
+
+OK, you are probably quite happy at this point, as our first rule is running as expected... However, we are not really done yet. Before playing our rule against any real projects, we have to finalize its creation within the custom plugin, by registering it.
+
+### Rule Metadata
+The first thing to do is to provide to our rule all the metadata which will allow us to register it properly in the SonarQube platform. To do so, add the `org.sonar.check.Rule` annotation to `MyFirstCustomCheck` class rule, and provide a **key**, a **name**, a **description** and optional **tags**, as in the following code snippet.
+
+```java
+@Rule(
+  key = "MyFirstCustomCheck",
+  name = "Return type and parameter of a method should not be the same",
+  description = "For a method having a single parameter, the types of its return value and its parameter should never be the same.",
+  priority = Priority.CRITICAL,
+  tags = {"bug"})
+public class MyFirstCustomCheck extends IssuableSubscriptionVisitor {
+  // ...
+}
+```
+
+### Rule Activation
+The second things to to is to activate the rule within the plugin. To do so, open class `RulesList` (`org.sonar.samples.java.RulesList`). In this class, you will notice methods `getJavaChecks()` and `getJavaTestChecks()`. These methods are used to register our rules with alongside the rule of the Java plugin. Note that rules registered in `getJavaChecks()` will only be played against source files, while rules registered in `getJavaTestChecks()` will only be played against test files. To register the rule, simply add the rule class to the list builder, as in the following code snippet:
+
+```java
+public static List<Class<? extends JavaCheck>> getJavaChecks() {
+  return ImmutableList.<Class<? extends JavaCheck>>builder()
```

Comment:
same, let's not use Guava APIs

Reply:
But for this I also need to modify the custom plugin template on the other repository.